src/v3/src/transformer/layout/consent/transformGranularConsent.ts

```diff
@@ -0,0 +1,118 @@
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  ButtonElement,
+  ButtonType,
+  DescriptionElement,
+  FieldElement,
+  IdxStepTransformer,
+  LinkElement,
+} from '../../../types';
+import { loc } from '../../../util';
+
+export const transformGranularConsent: IdxStepTransformer = ({ transaction, formBag }) => {
+  const { uischema } = formBag;
+  const {
+    // @ts-expect-error OKTA-598868 app is missing from rawIdxState type
+    rawIdxState: { app },
+    nextStep: { name: stepName } = {},
+  } = transaction;
+
+  // sorting field elements to display mutuable items first.
+  uischema.elements.sort((a, b) => {
+    if (a.type !== 'Field' || b.type !== 'Field') {
+      return 0;
+    }
+    const aField = a as FieldElement;
+    const bField = b as FieldElement;
+    if (aField.options.inputMeta.mutable === true && bField.options.inputMeta.mutable === false) {
+      return -1;
+    }
+    return 1;
+  });
+
+  const descriptionEle: DescriptionElement = {
+    type: 'Description',
+    contentType: 'subtitle',
+    noMargin: true,
+    options: {
+      content: loc('oie.consent.scopes.granular.description', 'login'),
+      dataSe: 'consent-description',
+    },
+  };
+
+  uischema.elements.unshift(descriptionEle);
+
+  const termsOfServiceHref = app?.value?.termsOfService
+    && (app.value.termsOfService as Record<string, Record<string, unknown>>).href;
+  if (termsOfServiceHref) {
+    const tosLink: LinkElement = {
+      type: 'Link',
+      options: {
+        label: loc('consent.required.termsOfService', 'login'),
+        href: termsOfServiceHref as string,
+        dataSe: 'terms-of-service',
+        target: '_blank',
+        step: '',
+      },
+    };
+    uischema.elements.push(tosLink);
+  }
+
+  const privacyPolicyHref = app?.value?.privacyPolicy
+    && (app.value.privacyPolicy as Record<string, Record<string, unknown>>).href;
+  if (privacyPolicyHref) {
+    const privacyPolicyLink: LinkElement = {
+      type: 'Link',
+      options: {
+        label: loc('consent.required.privacyPolicy', 'login'),
+        href: privacyPolicyHref as string,
+        dataSe: 'privacy-policy',
+        target: '_blank',
+        step: '',
+      },
+    };
+    uischema.elements.push(privacyPolicyLink);
+  }
+
+  const submitButton: ButtonElement = {
+    type: 'Button',
+    label: loc('consent.required.consentButton', 'login'),
+    options: {
+      type: ButtonType.BUTTON,
+      includeData: true,
+      actionParams: { consent: true },
+      step: stepName!,
```

Comment:
Same question here as in the other transformer about avoiding the non-null assertions.